### PR TITLE
Change Output Location List

### PR DESCRIPTION
**Pull request by andrew-mestas**
The content to be added to Changelog is as follows:
* 
* 
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@andrew-mestas__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (v-anmest). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by TianoMS** [comment]
@andrew-mestas There are build errors. Please take a look at the CI build log.

File: lib/commands/arm/location/location._js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    if (!options.json) {
+      log.warn('The "location list" commands is changed to list subscription\'s locations. ' + 
+      'For old information, use "provider list or show" commands.');
+      log.info('Getting locations...');
+    }
+
+
+    var subscription = profile.current.getSubscription(options.subscription);
+
+    var clientA = new resourceLib.SubscriptionClient(subscription._createCredentials(),
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: let's rename it to a meaningful name.

File: lib/commands/arm/location/location._js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  return result.map(function(x){ return {"id": x.id, "name": x.name, "displayName": x.displayName, "providers": getProviders(providers, x.displayName) }; });
+}
+
+function showList(providerArr, numToShow){ 
+  var smallerList = providerArr.slice(0, numToShow).join(", ");
+  var ellipse =  providerArr.length > 4 ? "..." : "";
+    return smallerList + ellipse;
+}
+
+function getProviders(locationData, name) {
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: rename it to "getProvidersForLocation"? Also, rename the parameters.

File: lib/commands/arm/location/location._js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  var providers = locationData.map(function(x) {
+    if (x.resourceTypes.filter(function(y){ return y.locations.filter(function(z){ return z == name }).length }).length > 0) {
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: do we want z === name?

File: lib/commands/arm/location/location._js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      return x.namespace;
+    }
+  }).filter(function(x){ return x != undefined });
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: can we do filter first and then map?

File: lib/commands/arm/location/location._js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            log.data($('DisplayName : '), location.displayName);
+            log.data($('Providers   : '), showList(location.providers, 4));
+          });
+        }
+      });
+  }));
+};
+
+function takeAwayLatLonAddProviders(result, providers) {
+  return result.map(function(x){ return {"id": x.id, "name": x.name, "displayName": x.displayName, "providers": getProviders(providers, x.displayName) }; });
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: put the arguments in separate lines.

File: lib/commands/arm/location/location._js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+function showList(providerArr, numToShow){ 
+  var smallerList = providerArr.slice(0, numToShow).join(", ");
+  var ellipse =  providerArr.length > 4 ? "..." : "";
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: How do the user get the full list of providers?

File: test/commands/arm/location/arm.location-tests.js [hunk 7]
```diff
@@ -49,6 +49,9 @@ describe('arm', function () {
           locations.some(function (location) {
             return location.name === 'westus' && location.displayName === 'West US';
           }).should.be.true;
+          locations.some(function (location) {
+            return location.providers.length > 1 && location.providers.indexOf("Microsoft.Batch") == 0;
```
Comment (TianoMS) [inline test/commands/arm/location/arm.location-tests.js]: let's do: ... && location.providers.indexOf("Microsoft.Batch") >= 0

File: lib/commands/arm/location/location._js (hunk 6 above)
Comment (andrew-mestas) [inline lib/commands/arm/location/location._js]: Users can append --json to the command in cli to get the full object.

**Comment by andrew-mestas** [comment]
All comments addressed in commits bcb063a and f0c2d9a.

File: lib/commands/arm/location/location._js [hunk 8]
```diff
@@ -83,13 +90,17 @@ function showList(providerArr, numToShow){
...
+      return resourceType.locations.filter(function(locationName){ 
+        return locationName === displayName; 
+      })
+    }).length > 0) {
+      return true;
     }
-  }).filter(function(x){ return x != undefined });
-return providers;
+  }).map(function(provider){ return provider.namespace });
+return providersByLocation;
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: indent here.

File: lib/commands/arm/location/location._js (hunk 6 above)
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: Not sure if we can do this (showing 4 items by default). Talk to Gaurav about it.

File: lib/commands/arm/location/location._js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+          providers = providerUtils.getAllProviders(client, _);
+        }, _);
+
+    if (!options.json) {
+      log.warn('The "location list" commands is changed to list subscription\'s locations. ' + 
+      'For old information, use "provider list or show" commands.');
+      log.info('Getting locations...');
+    }
+
+
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: remove extra space lines.

File: lib/commands/arm/location/location._js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    var subscription = profile.current.getSubscription(options.subscription);
+
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: remove this empty line also.

File: lib/commands/arm/location/location._js [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    var subscriptionClient = new resourceLib.SubscriptionClient(subscription._createCredentials(), 
+                                                   subscription.resourceManagerEndpointUrl);
+
+    var listOfLocations = subscriptionClient.subscriptions.listLocations(subscription.id, _);
+    listOfLocations = takeAwayLatLonAddProviders(listOfLocations, providers);
+
+      cli.interaction.formatOutput(listOfLocations, function (data) {
```
Comment (TianoMS) [inline lib/commands/arm/location/location._js]: indent.

**Pull request closed without merging**